### PR TITLE
Startup scripts

### PR DESCRIPTION
**Pull request by simeonschuez**
Added two scripts:

_start_server_and_bot.py_ starts slurk automatically connects bots specified as arguments. For example, the command to start the server and connect Pairup Bot and Multi Bot would be:
`python start_server_and_bot.py sample_bots/pairup_bot.py sample_bots/multi_bot.py`. 
If any virtual environment is needed to run slurk, the environment has to be activated before executing the script.

_start_clients.py_ automatically logs in clients to a running instance of slurk. It requests the required tokens, generates login URLs and opens them in web browsers specified in the script. By default, two clients are connected. The number of clients can be changed by passing more or less names to the function `get_client_links(*names, key)`.

Both scripts work for Linux and macOS.

### CODE REVIEW

**Comment by davidschlangen** [comment]
I think this is trying to be too clever / is doing too much. It doesn't play well with anaconda environments, as the new terminal that it opens will be in the default environment.

Can this not just use `subprocess()` to open the client, leaving it to the environment to ensure that `python` resolves to the right version?

**Comment by TimDiekmann** [comment]
I like the idea of a convenient startup script, but this is too specific. Things like browsers should be set in the config file (it is fine to add random things to the config, as they are not tracked).

Also, I think we shouldn't clutter the root folder even more. Maybe a `tools` directory?

**Comment by davidschlangen** [comment]
Good idea (`tools` directory). I can think of other stuff that could nicely go there as well. (Like an example jupyter-notebook for viewing logfiles in a convenient dialogue format.)

**Comment by simeonschuez** [comment]
I'll try and adapt the scripts until the end of the week, if that's okay.

**Comment by simeonschuez** [comment]
I updated the scripts:

- start_server_and_bot.py:

    - instead of running the scripts in new terminals, the subprocess module is now used (I wasn't completely sure how to properly terminate the subprocesses started in the scripts - for now, the process ids are stored in a list and the corresponding processes are terminated one by one if the script is stopped via CTRL+C)

- start_clients.py
    - the web browsers are now specified in the config.ini (section 'tools', keys: 'browser1' + 'browser2')
    - the client names are specified in the config.ini as well, separated by commas (section 'tools', key: 'client_names'). The number of client names determines the number of clients to be connected.
     - my config file (as an example): 

       [tools]
       browser1 = chromium-browser
       browser2 = firefox
       client_names = Dr. John Zoidberg,Professor Farnsworth

- everything moved to the tools directory

**Comment by TimDiekmann** [comment]
I'll look into this tomorrow or on tuesday if this is fine for you?

File: tools/start_server_and_bot.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    print ("Bots: ",bots, "\n")
+    processes = []
+
+    # start slurk
+    print ('starting server')
+    server = subprocess.Popen('python {path}/chat.py'.format(path=dir_path), shell=True, stdout=STDOUT, stderr=STDOUT)
+    # add process id to list of running processes
+    processes.append(server.pid)
+    # pause to allow server to start before adding bots
+    sleep(2)
```
Comment (TimDiekmann) [inline tools/start_server_and_bot.py]: Is it possible to actually *wait* for the server to start and give a feedback like this?:
```
starting server...
success
```

File: tools/start_server_and_bot.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+if __name__ == "__main__":
+    # print basic information
+    print ("Directory:",dir_path)
+    print ("Bots: ",bots, "\n")
+    processes = []
+
+    # start slurk
+    print ('starting server')
+    server = subprocess.Popen('python {path}/chat.py'.format(path=dir_path), shell=True, stdout=STDOUT, stderr=STDOUT)
```
Comment (TimDiekmann) [inline tools/start_server_and_bot.py]: Is it possible to forward `stderr` (and maybe `stdout`)?

File: tools/start_server_and_bot.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        bot_dir = dirname(i)
+        bot_filename = basename(i)
+        os.chdir(bot_dir)
+        token = get_bot_token(bot_filename, secret_key)
+
+        print ("\n\nstarting", i, "\ntoken:", token)
+        bot_process = subprocess.Popen('python {name} {bot_token}'.format(name=bot_filename, bot_token=token), shell=True, stdout=STDOUT, stderr=STDOUT)
+        # add process id to list of running processes
+        processes.append(bot_process.pid)
+        sleep(1)
```
Comment (TimDiekmann) [inline tools/start_server_and_bot.py]: Is it possible to actually _wait_ for the server to start and give a feedback like this?:

```
starting [bot path]...
using token [XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXX]
success
```

File: tools/start_server_and_bot.py (hunk 1 above)
Comment (TimDiekmann) [reply tools/start_server_and_bot.py]: Ii added an output to the server when it has been started

**Comment by simeonschuez** [comment]
There is now a feedback if the server was started successfully and if bots are connected. Server stdout and stderr are displayed in the terminal the startup script is called with. I tried to optionally redirect the stdout and stderr for bots to log files, but since this is not working reliably yet, it's not added to the scripts.

**Pull request merged**